File: ios/MMCoach/MMCoach/Models/RecentCaseRecord.swift
//
//  RecentCaseRecord.swift
//  MMCoach
//
//  Local-only metadata for the Home screen's "Recent Cases" list. The
//  backend does not currently expose a case-listing Cloud Function, so
//  MMCoach keeps a small local index of cases the trainee has started on
//  this device, refreshed from Back4App on demand for the case itself.
//

import Foundation

struct RecentCaseRecord: Codable, Identifiable, Hashable {
    let id: String
    var title: String
    let createdAt: Date
    var status: CaseStatus
}

File: android/app/src/main/java/dev/benderapps/mmcoach/models/RecentCaseRecord.kt
package dev.benderapps.mmcoach.models

import java.util.Date

/**
 * Local-only metadata for the Home screen's "Recent Cases" list. Mirrors
 * iOS `Models/RecentCaseRecord.swift` -- the backend has no case-listing
 * Cloud Function, so this stays a small local index refreshed on demand.
 */
data class RecentCaseRecord(
    val id: String,
    val title: String,
    val createdAt: Date,
    val status: CaseStatus,
)
